code/modules/projectiles/gun_system.dm

```diff
@@ -994,6 +915,419 @@ and you're good to go.
 /obj/item/weapon/gun/proc/delete_bullet(obj/projectile/projectile_to_fire, refund = FALSE)
 	return FALSE
 
+
+//----------------------------------------------------------
+				//							\\
+				//         RELOADING        \\
+				//							\\
+				//						   	\\
+//----------------------------------------------------------
+
+
+///Called after the gun fires. Handles the amunition side of firing.
+/obj/item/weapon/gun/proc/post_fire()
```

Comment:
Unneded proc, it's never overidden and it's called by Fire() and the flamer.

Inline this in Fire(), and just add the needed code in flamer. I'm not sure you even need code in flamer, flamer is calling finish_fire() as well?